src/platform/Darwin/PlatformManagerImpl.h

```diff
@@ -49,9 +50,8 @@ class PlatformManagerImpl final : public PlatformManager, public Internal::Gener
     dispatch_queue_t GetWorkQueue() { return mWorkQueue; }
     bool IsWorkQueueCurrentQueue() const;
 
-    CHIP_ERROR StartBleScan(BleScannerDelegate * delegate = nullptr);
+    CHIP_ERROR StartBleScan(BleScannerDelegate * delegate, BleScanMode mode = BleScanMode::kDefault);
```

Comment:
The invariant about how null delegate is allowed if mode is kPreWarm should be documented here, right?

That said, at this point do we have any non-delegate callers of this API?  If not, removing that ability in a follow-up would be good.

Reply:
I didn't come to any great conclusions on this... maybe I'll just remove the precondition for now. We currently handle null delegate in the code, and all delegate methods now have an empty default implementation, so functionally there isn't really a difference between null delegate and one that does nothing.